src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
@mindplay-dk ah I missed that in Diactoros. However, I still think the ergonomics of this change is correct. The problem with providing a file path as the target is two-fold:

1. It requires also setting a opening `mode`, which is what we see in Diactoros.
2. It is *one* function call for users to open a file and pass it as a resource and *at least two* function calls for users to create a resource from string content:

```php
// passing a file
$factory->createStream(fopen($file, 'r'));

// passing string of content
$resource = fopen('php://temp', 'r+');
fwrite($resource, $content);
$factory->createStream($resource);
```

And if we allow passing a string of content, the amount of code is significantly reduced:

```php
$factory->createStream($content);
```

This is why I suggested including a helper function that would allow for creating resources from strings of content in #11, as it would have changed the last example to:

```php
$factory->createStream(get_temporary_resource($content));
```

I'm not suggesting that #11 be reopened. I think this PR represents a better way forward from a user standpoint. PSR-7 implementations will need to have a bit of additional code, which seems like an acceptable trade off for increased user ergonomics.